app/controllers/api/submissions_controller.rb

```diff
@@ -8,14 +8,44 @@ class SubmissionsController < ApplicationController
     before_action :authenticate
 
     def create
-      result = { result: true, message: public_app_params['jwt'] }
-      render(json: result.to_json)
+      online_submission = OnlineApplicationBuilder.new(public_app_params).build
+      if online_submission.save
+        render(json: { result: true, message: online_submission.reference }.to_json)
```

Comment:
I think you can remove the `.to_json` method call as it's called automatically when rendered. 

Later we should probably encapsulate the successful / unsuccessful response to a presenter. But no need now.